vllm/model_executor/parameter.py

```diff
@@ -0,0 +1,191 @@
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+from torch.nn import Parameter
+
+from vllm.distributed import get_tensor_model_parallel_rank
+from vllm.logger import init_logger
+
+__all__ = [
+    "vLLMParameter", "PackedvLLMParameter", "PerTensorScaleParameter",
+    "ModelWeightParameter", "ChannelQuantScaleParameter",
+    "GroupQuantScaleParameter"
+]
+
+logger = init_logger(__name__)
+
+
+class vLLMParameter(Parameter):
```

Comment:
Can this be an abstract class that exposes:
- `load_column_parallel_weight`
- `load_qkv_weight`
- `load_merged_column_parallel_weight`
- `load_row_column_parallel_weight`

Reply:
I think no-ops over abstract. We want to return the loaded weight and parameter data unchanged in the base class

Reply:
See my updated note^

Perhaps we could just put the copying there?